cpp/dolfinx/common/log.cpp

```diff
@@ -6,3 +6,27 @@
 
 #include "log.h"
 #include "loguru.cpp"
+#include <vector>
+
+//-----------------------------------------------------------------------------
+void dolfinx::init_logging(int argc, char* argv[])
+{
+  loguru::g_stderr_verbosity = loguru::Verbosity_WARNING;
+
+#ifndef NDEBUG
+  loguru::SignalOptions signals;
+#else
+  loguru::SignalOptions signals = loguru::SignalOptions::none();
+#endif
```

Comment:
I found it a bit hard to read `if not defined "do not debug"`:
```suggestion
#ifdef NDEBUG
  loguru::SignalOptions signals = loguru::SignalOptions::none();
#else
  loguru::SignalOptions signals;
#endif
```